Sources/Datadog/RUM/RUMMonitor/Scopes/RUMResourceScope.swift

```diff
@@ -71,9 +71,9 @@ internal class RUMResourceScope: RUMScope {
         let eventData = RUMResourceEvent(
             date: resourceLoadingStartTime.timeIntervalSince1970.toMilliseconds,
             application: .init(id: context.rumApplicationID),
-            session: .init(id: context.sessionID.toString, type: "user"),
+            session: .init(id: context.sessionID.toRUMDataFormat, type: "user"),
```

Comment:
i see some `session`s take string and some take enum 🤔 i didn't understand how that compiles, do we have 2 different session type?

Reply:
I'm progressively removing manually created `RUM*Events` and replacing them with their `RUMDataModel` DTO counterparts. In this PR I removed `RUMErrorEvent` in favor of `RUMError`. In next PR I'm doing the same with `RUMResourceEvent` → `RUMResource`. Over the way we'll get rid of all hand-crafted models and it will look consistent 👌.